cleverhans/attacks.py

```diff
@@ -33,7 +34,23 @@ def __init__(self, model, back='tf', sess=None):
         self.model = model
         self.back = back
         self.sess = sess
-        self.inf_loop = False
+
+        # We are going to keep track of old graphs and cache them.
+        self.graphs = {}
```

Comment:
I'd make this an OrderedDict to reduce the amount of variability between different runs of the same script